cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go

```diff
@@ -87,7 +87,23 @@ type Etcd struct {
 	DataDir   string            `json:"dataDir"`
 	ExtraArgs map[string]string `json:"extraArgs,omitempty"`
 	// Image specifies which container image to use for running etcd. If empty, automatically populated by kubeadm using the image repository and default etcd version
-	Image string `json:"image"`
+	Image      string          `json:"image"`
+	SelfHosted *SelfHostedEtcd `json:"selfHosted,omitempty"`
+}
+
+// SelfHostedEtcd describes options required to configure self-hosted etcd
+type SelfHostedEtcd struct {
+	// CertificatesDir represents the directory where all etcd TLS assets are stored. By default this is
+	// a dir names "etcd" in the mains CertificatesDir value.
```

Comment:
nit: main

Reply:
Why not reuse the main certificatesdir completely?

Reply:
See my comment here: https://github.com/kubernetes/kubernetes/pull/54543#discussion_r147746299